spec/controllers/patient_cards_controller_spec.rb

```diff
@@ -1,54 +1,172 @@
+      it 'renders the :index view' do
+        expect(response).to render_template(:index)
+      end
+
+      it 'populates an array of patient cards' do
+        expect(assigns(:patients)).to eq([patient_card])
+      end
+    end
+
+    context 'Non-signed in physioterapist' do
+      before do
+        get :index
+      end
+
+      it 'redirects to sign in view' do
+        expect(response).to redirect_to(new_user_session_path)
+      end
     end
   end
-  #
-  # describe "GET #show" do
-  #   it "returns http success" do
-  #     get :show
-  #     expect(response).to have_http_status(:success)
-  #   end
-  # end
-
-  describe "GET #new" do
-    it "returns http success" do
-      get :new
-      expect(response).to have_http_status(:success)
+
+  describe 'GET #show' do
+    context 'Signed in physioterapist' do
+      before :each do
+        sign_in physio
+        get :show, params: { id: patient_card.id }
+      end
+      it 'returns http success' do
+        expect(response).to have_http_status(:success)
+      end
+
+      it 'renders the :show view' do
+        expect(response).to render_template(:show)
+      end
+    end
+    context 'Non-signed in physioterapist' do
+      before do
+        get :show, params: { id: patient_card.id }
+      end
+
+      it 'redirects to sign in view' do
+        expect(response).to redirect_to(new_user_session_path)
+      end
     end
   end
-  #
-  # describe "GET #create" do
-  #   it "returns http success" do
-  #     get :create
-  #     expect(response).to have_http_status(:success)
-  #   end
-  # end
-  #
-  # describe "GET #edit" do
-  #   it "returns http success" do
-  #     get :edit
-  #     expect(response).to have_http_status(:success)
-  #   end
-  # end
-  #
-  # describe "GET #update" do
-  #   it "returns http success" do
-  #     get :update
-  #     expect(response).to have_http_status(:success)
-  #   end
-  # end
-  #
-  # describe "GET #destroy" do
-  #   it "returns http success" do
-  #     get :destroy
-  #     expect(response).to have_http_status(:success)
-  #   end
-  # end
 
+  describe 'GET #new' do
+    context 'Signed in physioterapist' do
+      before :each do
+        sign_in physio
+        get :new
+      end
+      it 'returns http success' do
+        expect(response).to have_http_status(:success)
+      end
+
+      it 'renders the :new view' do
+        expect(response).to render_template(:new)
+      end
+    end
+
+    context 'Non-signed in physioterapist' do
+      before do
+        get :new
+      end
+
+      it 'redirects to sign in view' do
+        expect(response).to redirect_to(new_user_session_path)
+      end
+    end
+  end
+
+  describe 'POST #create' do
+    context 'Signed in physioterapist' do
+      before :each do
+        sign_in physio
+      end
+      context 'with valid attributes' do
+        it 'creates new patient card' do
+          expect do
+            post :create, params: { patient_card: {
```

Comment:
firstly create let with params for post, like:
```
let(:params) do
{ patient_card: {
     first_name: 'Jan',
     last_name: 'Nowak',
     date_of_birth: Time.now - 48.years,
      phone_number: 123_123_123,
      comments: 'Lorem ipsum'
} }
```
and create let or subject for post action:
```
let(:post_action) do
  post :create, params: params
end
```
Then u can use this action multiple times in its by writing sth like
```
expect { post_action }.to change(PatientCard, :count).by(1)
```
Remember to not use ! after this lets, cos it creates action immediately and we don't want that here